snap/snapcraft.yaml

```diff
@@ -85,43 +80,58 @@ parts:
-      install -DT "./cmd/device-rfid-llrp" "$SNAPCRAFT_PART_INSTALL/bin/device-rfid-llrp"
-
-      install -d "$SNAPCRAFT_PART_INSTALL/config/device-rfid-llrp/res/"
-      install -DT "./cmd/res/configuration.toml" \
-        "$SNAPCRAFT_PART_INSTALL/config/device-rfid-llrp/res/configuration.toml"
-
-      install -d "$SNAPCRAFT_PART_INSTALL/config/device-rfid-llrp/res/profiles/"
-      for fpath in ./cmd/res/profiles/*.yaml; do
-          fname=$(basename "$fpath")
-          install -DT "./cmd/res/profiles/${fname}" \
-            "$SNAPCRAFT_PART_INSTALL/config/device-rfid-llrp/res/profiles/${fname}"
-      done
-
-      install -d "$SNAPCRAFT_PART_INSTALL/config/device-rfid-llrp/res/provision_watchers/"
-      for fpath in ./cmd/res/provision_watchers/*.json; do
-          fname=$(basename "$fpath")
-          install -DT "./cmd/res/provision_watchers/${fname}" \
-            "$SNAPCRAFT_PART_INSTALL/config/device-rfid-llrp/res/provision_watchers/${fname}"
-      done
-
-      install -DT "./Attribution.txt" \
-         "$SNAPCRAFT_PART_INSTALL/usr/share/doc/device-rfid-llrp/Attribution.txt"
-      install -DT "./LICENSE" \
-         "$SNAPCRAFT_PART_INSTALL/usr/share/doc/device-rfid-llrp/LICENSE"
-
-  config-common:
+      install -DT ./cmd/device-rfid-llrp "$SNAPCRAFT_PART_INSTALL/bin/device-rfid-llrp"
+      install -DT ./Attribution.txt "$SNAPCRAFT_PART_INSTALL/usr/share/doc/device-rfid-llrp/Attribution.txt"
+      install -DT ./LICENSE "$SNAPCRAFT_PART_INSTALL/usr/share/doc/device-rfid-llrp/LICENSE"
+
+      LLRP_RES_DIR="$SNAPCRAFT_PART_INSTALL/config/device-rfid-llrp/res/"
+      mkdir -p "$LLRP_RES_DIR/profiles"
+      mkdir -p "$LLRP_RES_DIR/devices"
+      mkdir -p "$LLRP_RES_DIR/provision_watchers"
+      cp ./cmd/res/configuration.toml "$LLRP_RES_DIR/"
+      cp ./cmd/res/profiles/*.yaml "$LLRP_RES_DIR/profiles/"
+      cp ./cmd/res/provision_watchers/*.json "$LLRP_RES_DIR/provision_watchers/"
+
+
+  runtime-helpers:
     plugin: dump
     source: snap/local/runtime-helpers
+
+  metadata:
+    plugin: nil
+    source: https://github.com/canonical/edgex-snap-metadata.git
+    source-branch: appstream
+    source-depth: 1
+    override-build: |
+      # install the icon at the default internal path
+      install -DT edgex-snap-icon.png \
+        $SNAPCRAFT_PART_INSTALL/meta/gui/icon.png      
+      # change to this project's repo to get the version
+      cd $SNAPCRAFT_PROJECT_DIR
+      if git describe ; then
+        VERSION=$(git describe --tags --abbrev=0 | sed 's/v//')
+      else
+        VERSION="0.0.0"
+      fi
+      
+      # write version to file for the build
+      echo $VERSION > ./VERSION
+      # set the version of this snap
+      snapcraftctl set-version $VERSION
+    parse-info: [edgex-device-rfid-llrp.metainfo.xml]
+
+ 
+  # copy auto-configure.sh to /bin and make sure curl is installed
+  auto-configure:
+    plugin: dump
+    source: bin/
+    organize:
+      auto-configure.sh: bin/auto-configure.sh
```

Comment:
This also copies `bin/test-attribution-txt.sh` and leaves it under $SNAP. Need to include only `bin/auto-configure.sh`, or if possible just source a single file `bin/auto-configure.sh`.

Reply:
pushed an update with
```
  auto-configure:
    plugin: dump
    source: bin/
    filesets:
      auto-configure: [bin/auto-configure.sh]
    organize:
      auto-configure.sh: bin/auto-configure.sh
    stage-packages:
      - curl
 ```
which just copies this file